### DISCUSSION TITLE
How to handle IPSpoofing

### DISCUSSION THREAD

**Discussion by pravsjv** [keycloak/keycloak#28445] [Q&A]
Lets assume we have a user from a rouge state who has gotten some stolen keycloak credentials, This person connects to a VPN address space and connects to network device. In this we need to identify the user's IP Address and not the VPN address. 

Does anyone know if there is a way to do it in keycloak?

**Comment by codespearhead** [comment]
No, you can't. That's the point of VPNs.

If Keycloak is compromised, the best course of action is to regenerate all the credentials and prevent the behavior that allowed it to happen from happening again, such as enforcing 2FA if we're talking about a user.

What you currently can do, however, is to check the logs of the VPN server for the IPs that connected to it.
